Add documentation for nodes api (#1560)
opensearch-project/documentation-website: _opensearch/rest-api/nodes-apis/nodes-reload-secure.md

---
layout: default
title: Nodes reload secure settings
parent: Nodes APIs
grand_parent: REST API reference
nav_order: 50
---

# Nodes reload secure settings

The nodes reload secure settings endpoint allows you to change secure settings on a node and reload the secure settings without restarting the node.

## Path and HTTP methods

```
POST _nodes/reload_secure_settings
POST _nodes/<node id>/reload_secure_settings
```

## Path parameter

> You can include the following optional path parameter in your request:

Reviewer: (suggestion): if you only have one parameter, you can remove the table and just define it like this:
"You can specify individual node names you want to reload. Provide the `node_id` (type: String) in your request.

Author: I think it will be good to leave it as a table for visual consistency between pages.

Parameter | Type | Description
:--- | :--- | :---

> node_id | String | The names of the nodes to reload.

Reviewer 2: ```suggestion
node_id | String | nodeId | String | A comma-separated list of nodeIds to filter results. Supports [node filters]({{site.url}}{{site.baseurl}}/opensearch/rest-api/nodes-apis/index/#node-filters). Defaults to `_all`.
```

Reviewer 2: I don't know if we've reached consensus on how to format parameters in angle brackets, but other nodes APIs use <nodeId>. We can change that, but we need to make it consistent across all nodes APIs.

> ## HTTP request body

Reviewer 2: ```suggestion
## Request fields
```

> An optional object containing the password for the OpenSearch keystore.

Reviewer 2: ```suggestion
The request may include an optional object containing the password for the OpenSearch keystore.
```

````json
{

>   "secure_settings_password":"keystore_password"

Reviewer 2: ```suggestion
  "secure_settings_password": "keystore_password"
```

}
````

> ## Required permissions

Reviewer 2: Could we move this to be the last section? That's what we have in the other nodes APIs.

If you use the security plugin, make sure you set the following permissions: `cluster:manage/nodes`.

> ### Example API query and response

Reviewer 2: ```suggestion
#### Sample request
```

The following is an example API query:

```
POST _nodes/reload_secure_settings
```

> 

Reviewer 2: Add #### Sample response here.

---
layout: default
title: Nodes reload secure settings
parent: Nodes APIs
grand_parent: REST API reference
nav_order: 50
---

# Nodes reload secure settings

The nodes reload secure settings endpoint allows you to change secure settings on a node and reload the secure settings without restarting the node.

## Path and HTTP methods

```
POST _nodes/reload_secure_settings

> POST _nodes/<node id>/reload_secure_settings

Reviewer 2: ```suggestion
POST _nodes/<nodeId>/reload_secure_settings
```

```

## Path parameter

> You can include the following optional path parameter in your request:

Reviewer 3: Replace the colon with a period (don't introduce tables with a colon).

Parameter | Type | Description
:--- | :--- | :---

> nodeId | String | A comma-separated list of nodeIds to filter results. Supports [node filters]({{site.url}}{{site.baseurl}}/opensearch/rest-api/nodes-apis/index/#node-filters). Defaults to `_all`.

Reviewer 3: Change to "A comma-separated list of nodeIds used to filter results" (add "used"). In the next file, we use "node IDs" (instead of "nodeIds"). Let's make consistent.
